src/modules/commander/Commander.cpp

```diff
@@ -4058,6 +4066,156 @@ void Commander::battery_status_check()
 	}
 }
 
+void Commander::airspeed_use_check()
+{
+	// assume airspeed sensor is good before starting FW flight
+	bool valid_flight_condition = (status.arming_state == vehicle_status_s::ARMING_STATE_ARMED) &&
+			!status.is_rotary_wing &&
+			!land_detector.landed;
+	bool fault_declared = false;
+	bool fault_cleared = false;
+	if (!valid_flight_condition) {
+		_tas_use_inhibit = false;
+		_time_tas_good_declared = hrt_absolute_time();
+		_time_tas_bad_declared = 0;
+		status.aspd_check_failing = false;
+		status.aspd_fault_declared = false;
+		status.aspd_use_inhibit = false;
+		status.aspd_fail_rtl = false;
+		_time_last_airspeed = hrt_absolute_time();
+		_airspeed_fault_type = new(char[7]);
```

Comment:
This will leak memory by allocating a new char array every iteration with invalid flight conditions.

Reply:
Can you please provide recommendations on how to fix it. I do not normally deal with character arrays.

Reply:
Have fixed in rebased version.